src/commands/copy.rs

```diff
@@ -89,69 +91,88 @@ impl CopyCmd {
                 log!(level, "{}", merge_log);
             }
             let target_opt = &target_config.repository;
-
-            let repo_dest = get_repository(target_opt)?;
-
-            info!("copying to target {}...", repo_dest.name);
-            let repo_dest = if self.init && repo_dest.config_id()?.is_none() {
-                if config.global.dry_run {
-                    error!(
-                        "cannot initialize target {} in dry-run mode!",
-                        repo_dest.name
-                    );
-                    continue;
-                }
-                let mut config_dest = repo.config().clone();
-                config_dest.id = Id::random();
-                let pass = init_password(&repo_dest)?;
-                repo_dest.init_with_config(&pass, &self.key_opts, config_dest)?
-            } else {
-                open_repository(target_opt)?
-            };
-
-            if poly != repo_dest.config().poly()? {
-                bail!("cannot copy to repository with different chunker parameter (re-chunking not implemented)!");
+            let hooks = target_config.global.hooks.with_context("target");
+            if let Err(err) = self.copy(&repo, target_opt, &snapshots, &hooks) {
+                error!("error copying to target: {err}");
+                hooks.run_failed()?;
             }
+            hooks.run_finally()?;
```

Comment:
My understanding: we apply "global" hooks for a 2nd (target) repository.
But we do it from the command implementation. I am thinking this a bit tricky, as the link between "opening a 2nd repo with some `target` config" and "applying the global hooks from the `target` config" exist in the `CopyCmd` implementation.

In an ideal world, this link would rather be defined in some `type` or factorized `function` that handles anything related with an extra `target` repository.

My fear: tomorrow someone implements some other command that interacts with a 2nd `target` repo and forgets to use the `global.hooks` which could lead to a bug.

I might be over-anticipating, let me know what you about your opinion on that!

Reply:
Yes, you are perfectly right!
Actually the `copy` command so far is the only use case for multiple repositories, but there could be more.

So far the global hooks are not implemented when the repository is opened or closed, which should be the right place...

What do you think about adding even one more hooks section: `[repository.hooks]`. These would be hooks which are run when accessing a repository. In contrast, the `[global.hooks]` would be run for every command (even `self-update` or `show-config`). 

In this setting `[global.hooks]` would be mainly used to start log-like services, e.g. sending an email for failed runs, whereas `[repository.hooks]` would be used to define pre- and post-actions for repository access.

Then, the `copy` command would only call the `[repository.hooks]`.

Reply:
[repository.hooky] is now added.